src/bin/console/console.cpp

```diff
@@ -636,51 +667,74 @@ int Console::_visualize(const std::string& input) {
   const auto img_filename = plan_type_str + ".png";
 
   // Visualize the Logical Query Plan
-  if (plan_type == PlanType::LQP || plan_type == PlanType::UnoptLQP) {
-    std::vector<std::shared_ptr<AbstractLQPNode>> lqp_roots;
-
-    try {
-      if (!no_execute) {
-        // Run the query and then collect the LQPs
-        _sql_pipeline->get_result_table();
+  switch (plan_type) {
+    case PlanType::LQP:
+    case PlanType::UnoptLQP: {
+      std::vector<std::shared_ptr<AbstractLQPNode>> lqp_roots;
+
+      try {
+        if (!no_execute) {
+          // Run the query and then collect the LQPs
+          _sql_pipeline->get_result_table();
+        }
+
+        const auto& lqps = (plan_type == PlanType::LQP) ? _sql_pipeline->get_optimized_logical_plans()
+                                                        : _sql_pipeline->get_unoptimized_logical_plans();
+        for (const auto& lqp : lqps) {
+          lqp_roots.push_back(lqp);
+        }
+      } catch (const std::exception& exception) {
+        out(std::string(exception.what()) + "\n");
+        _handle_rollback();
+        return ReturnCode::Error;
       }
 
-      const auto& lqps = (plan_type == PlanType::LQP) ? _sql_pipeline->get_optimized_logical_plans()
-                                                      : _sql_pipeline->get_unoptimized_logical_plans();
-      for (const auto& lqp : lqps) {
-        lqp_roots.push_back(lqp);
+      LQPVisualizer visualizer;
+      visualizer.visualize(lqp_roots, graph_filename, img_filename);
+    } break;
+
+    case PlanType::PQP: {
+      // Visualize the Physical Query Plan
+      SQLQueryPlan query_plan{CleanupTemporaries::No};
+
+      try {
+        if (!no_execute) {
+          _sql_pipeline->get_result_table();
+        }
+
+        // Create plan with all roots
+        const auto& plans = _sql_pipeline->get_query_plans();
+        for (const auto& plan : plans) {
+          query_plan.append_plan(*plan);
+        }
+      } catch (const std::exception& exception) {
+        out(std::string(exception.what()) + "\n");
+        _handle_rollback();
+        return ReturnCode::Error;
       }
-    } catch (const std::exception& exception) {
-      out(std::string(exception.what()) + "\n");
-      _handle_rollback();
-      return ReturnCode::Error;
-    }
 
-    LQPVisualizer visualizer;
-    visualizer.visualize(lqp_roots, graph_filename, img_filename);
+      SQLQueryPlanVisualizer visualizer;
+      visualizer.visualize(query_plan, graph_filename, img_filename);
+    } break;
 
-  } else {
-    // Visualize the Physical Query Plan
-    SQLQueryPlan query_plan{CleanupTemporaries::No};
+    case PlanType::Joins: {
+      out("NOTE: Only Inner and Cross-Joins are visualized at the moment.\n");
```

Comment:
Can we somehow identify cases in which edges are missing and be more specific in the warning?

Reply:
No edges "missing", we just don't show Outer Joins. Adjusted the NOTE to reflect this.

"NOTE: Join graphs will show only Cross and Inner joins, not Outer joins"

Reply:
Can we somehow identify outer joins and name them in the warning instead of generally referring to "outer joins"? ;)

Reply:
Not 100% sure what you mean. I changed the warning to "NOTE: Join graphs will show only Cross and Inner joins, not Semi, Left, Right, Outer and Anti joins", does that come closer to what you want? It is out of the question right now to include non-Inner/Cross joins in the JoinGraph, DPccp cannot handle them.